### DISCUSSION TITLE
How to only create a file in a template without simultaneously rendering the file content in current template?

### DISCUSSION THREAD

**Discussion by lakca** [kubb-labs/kubb#1170] [Q&A]
As following, the file content will also render in the client default template.
```ts
export const clientTemplates = {
  ...Client.templates,
  default: function (props: React.ComponentProps<typeof Client.templates.default>) {
    const { name, generics, returnType, params, JSDoc, client } = props
    const { pluginManager } = useApp<PluginClient>()
    const file = pluginManager.getFile({ name: `${name}.constant`, extName: '.ts', pluginKey: ['plugin-client'] })
    return (
      <>
        <File baseName={file.baseName} path={file.path} meta={file.meta} override>
          <Text indentSize={0}>
            /** Path */
            <br/>
            export const {name}$Path = '{client.path.toURLPath()}'
            <br/>
            <br/>
            /** Method */
            <br/>
            export const {name}$Method = '{client.method.toUpperCase()}'
            <br/>
            <br/>
            /** ContentType */
            <br/>
            export const {name}$ContentType = '{client.contentType}'
            <br/>
            <br/>
          </Text>
        </File>
        <Client.templates.default {...props}/>
    </>
  )
  },
} as const
```

**Comment by stijnvanhulle** [comment]
You also render '<Client.templates.default {...props}/>' again, is this needed? Because that part will again render the default template. Btw I'm working on some docs for templates, feel free to add some comments on what we can improve here

**Reply by lakca** [reply]
Sorry, late reply. 
Yes，the default is needed. I want to additionally render a file to providing constants.

**Reply by lakca** [reply]
Or, maybe providing a way to directly add more `generators`.

But I think it's better to give an attribute (likes `fileOnly`) for `<File>` to instruct not render(conditional return) its content in current dom, since we can use some calculated parameters in current component.

**Reply by stijnvanhulle** [reply]
@lakca The templates are not always easy so in v3 of Kubb they will be removed. I'm thinking about a new api that would not need any React

**Reply by lakca** [reply]
Okay, appreciate your effort.
